### PR TITLE
Fix the formula for RedHat based OSes and clean the code

### PR DESCRIPTION
**Pull request by porunov**
This PR fixes the situation when you try to update Hadoop to the newer version. 
The problem is that the alternative is already installed with 30 priority. It's not possible to add another alternative with the same priority. That is why salt will show an error.
This PR skips alternative installation if the one is already installed, but it changes symlinks. 
Also, I've cleaned a little bit of code for better readability.

Personally, I don't understand why we need alternatives in this case because the alternatives won't be added in any cases (only one alternative will live except the situation when user add another one), so I would remove all alternatives and let only symlinks. But I am not sure, maybe somebody uses it for other purposes which I don't realize.

This PR isn't connected to the multi-cluster deployment and Hadoop 3. Those PRs will be added after this one.

### CODE REVIEW

**Comment by porunov** [comment]
@sroegner Hello,
Could you check this PR please?

**Comment by porunov** [comment]
Additionally, I found that some environment variables are missing. Also, some environment variables aren't set after the installation. Fixes provided in the above commit.

**Comment by sroegner** [comment]
@porunov this looks like a very nice cleanup. In general I would have been much more comfortable with this being several smaller PRs but I see how this developed.

Please tell me you test-run this code?

**Comment by porunov** [comment]
@sroegner yes, but only on CentOS 7.3 and not all functionality. I am testing all states right now.
After this PR I will send smaller PRs which affect less files.
I think the PR can wait till Monday / Tuesday. I will say when the testing is done.

**Comment by sroegner** [comment]
@porunov I very much appreciate your making this extra testing effort

**Comment by porunov** [comment]
@sroegner all functionality are successfully tested. PR can be merged. Check if you see something you don't like and tell me. I will change it if so. 
I am ready to provide additional PRs when this PR is merged.

**Comment by porunov** [comment]
@sroegner , Last commit adds a possibility to use your own fencing method in case when the user don't want to use the default one.

**Comment by blbradley** [comment]
@sroegner I also feel uncomfortable about such a large change at once. Not saying it shouldn't be merged, just wanted to validate your feelings.

**Comment by porunov** [comment]
@sroegner
I am ready to provide hadoop3 support for the formula. 
Please check this PR and merge it if everything is ok.
After that I will make a new pull request.

**Comment by sroegner** [comment]
@blbradley Completely agree - we discussed this above but at this point we'll have to move forward and try to minimize risk by testing.

@porunov Please if you can help it stop adding commits to this

File: hadoop/conf/hdfs/core-site.xml [hunk 1]
```diff
@@ -6,43 +6,27 @@
...
         <value>{{ hdfs.tmp_dir }}</value>
     </property>
 -->
-{%- else %}
+
     <property>
-      <name>fs.defaultFS</name>
-      <value>hdfs://{{hdfs.ha_cluster_id}}</value>
+        <name>{{ 'fs.default.name' if major == '1' else 'fs.defaultFS' }}</name>
+        <value>hdfs://{{ hdfs.namenode_host+':'+hdfs.namenode_port if hdfs.namenode_count == 1 else hdfs.ha_cluster_id }}</value>
```
Comment (sroegner) [inline hadoop/conf/hdfs/core-site.xml]: This breaks wherever one has specified the namenode port as an integer - please revert the port back to being its own expression

**Comment by porunov** [comment]
@sroegner sure, no unneeded commits will be added to this PR.
I fixed that expression. Tell me if something else must be fixed.

**Pull request merged**